cfme/infrastructure/provider/__init__.py

```diff
@@ -388,3 +388,19 @@ def wait_for_a_provider():
     wait_for(lambda: int(view.entities.paginator.items_amount), fail_condition=0,
              message="Wait for any provider to appear", num_sec=1000,
              fail_func=view.browser.refresh)
+
+
+def wait_for_infra_relationship_refresh(provider):
+    view = navigate_to(provider, 'Details')
+
+    def _check():
+        if (view.entities.summary("Status").read()['_table']['Last Refresh'] ==
+                'Success - Less Than A Minute Ago'):
+            return True
+        return False
+
+    logger.info('Waiting for relationship refresh')
+    wait_for(lambda: _check(),
+             delay=15,
+             timeout=110,
+             fail_func=view.browser.refresh)
```

Comment:
**REQUIRED**
These two functions are identical. Please put one it in cfme/tests/cloud_infra_common/test_relationships.py and delete another.

Reply:
I think as @rlbabyuk  suggested I should place this in cfme/common/provider.py which is a model page, as this wait_for can be used by other test cases as well.

Reply:
@anurag03  I think it's better to put in `cfme/tests/cloud_infra_common/test_relationships.py`, because that function ise used only there

Reply:
@quarckster I am adding the wait for in cfme/cloud_infra_common/test_relationships.py, but I still feel it should be a part of model page.